tests/dtypes/test_required.py

```diff
@@ -0,0 +1,224 @@
+    ])
+
+    with pytest.raises(Exception) as e:
+        app.post('/example/City', json={
+            'description': 'Test city',
+        })
+    assert e.value.args[0] == (
+        "'name' is required for 'example/City'."
+    )
+    resp = app.post('/example/City', json={
+        'name': 'City',
+        'description': 'Test city',
+    })
+    assert resp.status_code == 201
+    resp = app.get('/example/City')
+    assert listdata(resp, full=True) == [
+        {
+            'name': "City",
+            'description': 'Test city',
+        }
+    ]
+
+
+def test_update_with_required_property(
+    rc: RawConfig,
+    postgresql: str,
+    request: FixtureRequest,
+):
+    context = bootstrap_manifest(rc, '''
+    d | r | b | m | property          | type            | ref
+    example                           |                 |
+      |   |   | City                  |                 |
+      |   |   |   | name              | string required |
+      |   |   |   | description       | string          |
+    ''', backend=postgresql, request=request)
+
+    app = create_test_client(context)
+    app.authmodel('example/City', [
+        'insert',
+        'update',
+        'getall',
+    ])
+
+    resp = app.post('/example/City', json={
+        'name': 'City',
+        'description': 'Test city',
+    })
+    city = resp.json()
+
+    with pytest.raises(Exception) as e:
+        app.put(f'/example/City/{city["_id"]}', json={
+            '_revision': city["_revision"],
+            'description': 'Test city updated',
+        })
+    assert e.value.args[0] == (
+        "'name' is required for 'example/City'."
+    )
+    resp = app.put(f'/example/City/{city["_id"]}', json={
+        '_revision': city["_revision"],
+        'name': 'City',
+        'description': 'Test city updated',
+    })
+    assert resp.status_code == 200
+    resp = app.get('/example/City')
+    assert listdata(resp, full=True) == [
+        {
+            'name': "City",
+            'description': 'Test city updated',
+        }
+    ]
+
+
+def test_patch_with_required_property(
+    rc: RawConfig,
+    postgresql: str,
+    request: FixtureRequest,
+):
+    context = bootstrap_manifest(rc, '''
+    d | r | b | m | property          | type            | ref
+    example                           |                 |
+      |   |   | City                  |                 |
+      |   |   |   | name              | string required |
+      |   |   |   | description       | string          |
+    ''', backend=postgresql, request=request)
+
+    app = create_test_client(context)
+    app.authmodel('example/City', [
+        'insert',
+        'patch',
+        'getall',
+    ])
+
+    resp = app.post('/example/City', json={
+        'name': 'City',
+        'description': 'Test city',
+    })
+    city = resp.json()
+
+    resp = app.patch(f'/example/City/{city["_id"]}', json={
+        '_revision': city["_revision"],
+        'description': 'Test city updated',
```

Comment:
```suggestion
        'name': None,
        'description': 'Test city updated',
```

This would be a better test to check required.